aio/tools/transforms/angular-base-package/post-processors/auto-link-code.js

```diff
@@ -30,51 +30,64 @@ module.exports = function autoLinkCode(getDocFromAlias) {
   function autoLinkCodeImpl() {
     return (ast, file) => {
       visit(ast, 'element', (node, ancestors) => {
-        // Only interested in code elements that:
-        // * do not have `no-auto-link` class
-        // * do not have an ignored language
-        // * are not inside links
-        if (autoLinkCodeImpl.codeElements.some(elementType => is(node, elementType)) &&
-            (!node.properties.className || !node.properties.className.includes('no-auto-link')) &&
-            !autoLinkCodeImpl.ignoredLanguages.includes(node.properties.language) &&
-            ancestors.every(ancestor => !is(ancestor, 'a'))) {
-          visit(node, 'text', (node, ancestors) => {
-            // Only interested in text nodes that are not inside links
-            if (ancestors.every(ancestor => !is(ancestor, 'a'))) {
-              const parent = ancestors[ancestors.length - 1];
-              const index = parent.children.indexOf(node);
+        if (!isValidCodeElement(node, ancestors)) {
+          return;
+        }
 
-              // Can we convert the whole text node into a doc link?
-              const docs = getDocFromAlias(node.value);
-              if (foundValidDoc(docs, node.value, file)) {
-                parent.children.splice(index, 1, createLinkNode(docs[0], node.value));
-              } else {
-                // Parse the text for words that we can convert to links
-                const nodes =
-                    textContent(node)
-                        .split(/([A-Za-z0-9_.-]+)/)
-                        .filter(word => word.length)
-                        .map((word, index, words) => {
-                          // remove docs that fail the custom filter tests
-                          const filteredDocs = autoLinkCodeImpl.customFilters.reduce(
-                              (docs, filter) => filter(docs, words, index), getDocFromAlias(word));
-                          return foundValidDoc(filteredDocs, word, file) ?
-                              // Create a link wrapping the text node.
-                              createLinkNode(filteredDocs[0], word) :
-                              // this is just text so push a new text node
-                              {type: 'text', value: word};
-                        });
+        visit(node, 'text', (node, ancestors) => {
+          // Only interested in text nodes that are not inside links
+          const isInsideLink = ancestors.every(ancestor => !is(ancestor, 'a'));
+          if (!isInsideLink) {
```

Comment:
```suggestion
          const isInsideLink = ancestors.some(ancestor => is(ancestor, 'a'));
          if (isInsideLink) {
```

BTW, since this same logic is used in `isValidCodeElement()`, you could extract it to a standalone function.

Reply:
Is the proposed reverted logic line 40 correct? Exit out if we found a link? I thought the logic should run only for links. 😕

Reply:
This post-processor generates links from code inside code-blocks. We do not want to generate any links inside other links.

So, yes, the proposed change in my previous comment is correct.

Reply:
Sounds good